From 4b4600529b0b38d51a115ac0a9eab77e5370cb07 Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Wed, 10 Jul 2019 10:36:24 +0800
Subject: [PATCH] Fix kernel panic on suspend/resume and rmmod of btusb sco
 driver

Fixes for kernel panic seen on rmmod
    - URBs freed
    - Sound card is disconnected
    - Release USB interfaces

Fixes for kernel panic on suspend/resume
    - Provide a dummy implementation for suspend/resume

- Enable BT SCO HCI driver

Tracked-On: OAM-80746
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 kernel_config/kernel_64_defconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel_config/kernel_64_defconfig b/kernel_config/kernel_64_defconfig
index 8c8cb1e..c8deb71 100644
--- a/kernel_config/kernel_64_defconfig
+++ b/kernel_config/kernel_64_defconfig
@@ -1555,7 +1555,6 @@ CONFIG_NVME_TARGET=y
 CONFIG_MISC_RTSX=y
 CONFIG_UID_SYS_STATS=y
 # CONFIG_UID_SYS_STATS_DEBUG is not set
-CONFIG_BT_SCOHCI=m
 # CONFIG_C2PORT is not set
 
 #
@@ -4212,6 +4211,7 @@ CONFIG_SND_USB_HIFACE=m
 # CONFIG_SND_USB_PODHD is not set
 # CONFIG_SND_USB_TONEPORT is not set
 # CONFIG_SND_USB_VARIAX is not set
+CONFIG_BT_SCOHCI=m
 CONFIG_SND_SOC=y
 CONFIG_SND_SOC_AC97_BUS=y
 # CONFIG_SND_SOC_AMD_ACP is not set
-- 
2.21.0

